packages/server/assets/workspacesCore/typedefs/workspaces.graphql

```diff
@@ -28,7 +28,10 @@ input WorkspaceUpdateInput {
 input WorkspaceRoleUpdateInput {
   userId: String!
   workspaceId: String!
-  role: WorkspaceRole!
+  """
+  Leave role null to revoke access entirely
+  """
+  role: WorkspaceRole
```

Comment:
The convention so far (e.g. for projects) has been to return the real/internal role string (e.g. "workspace:admin"). The main reason for these secondary GQL enums is just to have GQL level validation/type checking for input objects - e.g. when updating a role or whatever.

It is a bit confusing, I know haha. I'd be open to revisiting these role enums and possibly dumping them in favor of just always using the real role values.

Reply:
noted, definitely got these two twisted a few times while working too. but will do this going forwards